### DISCUSSION TITLE
vdialog example

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#313] [Q&A] [answered]
Hi,

Is there an example on how to make a VDialog work? I have a button "open dialog" which goes to `def show_dialog():`
What would I need to do to get a popup dialog window with a simple text and a close button? I was trying to create a structure like below, but I could not make it produce a dialog window :
```python
def show_dialog():
    with vuetify.VDialog(..):
      with vuetify.VCard(..):
        with vuetify.VBtn(..)

```

**Comment by jourdain** [comment, marked as answer]
Ideally with trame your ui is "static" and you let vue handle its reactivity. 

So you need to put your dialog with your layout definition. 

```python
def show_dialog():
    state.show_dialog = True

with layout:
   # ... normal ui
   with vuetify.Dialog(v_model=("show_dialog", False):
        # ....
```

**Reply by bigfooted** [reply]
Oh great thanks, so everything is already there, you just change the visibility! Ok, I got it working using that logic:

```python
def update_dialog():
  state.show_dialog = not state.show_dialog

# ...

  with layout.content:
    # ...
    vuetify.VBtn("open dialog", color="green", click=update_dialog)
      with vuetify.VDialog(v_model=("show_dialog",False)):
        with vuetify.VCard():
          vuetify.VCardTitle("dialog window")
          with vuetify.VCardText():
              vuetify.VBtn("close", click=update_dialog)

```

Thanks again!

**Reply by jourdain** [reply]
No problem. 

BTW, you don't need a Python function to toggle the visibility. 

You can just do 

```python
vuetify.VBtn("close", click="show_dialog = !show_dialog")
```
